src/server/ws-routes/controllers/Friends.js

```diff
@@ -0,0 +1,121 @@
+const namespaceInfo = require('../../protocol/protocol.json').friends
+const EVENT_TYPES = namespaceInfo.eventTypes
+const logger = require('../../logger')
+const _ = require('lodash')
+class Friends {
+	constructor({
+		UserModel = require('../../models/user')
+	} = {}) {
+		this.UserModel = UserModel
+	}
+
+	[EVENT_TYPES.CONNECTION](socket, connections) {
+		const username = socket.request.user.username
+		connections.usersToConnectionsMap.set(username, socket)
+	}
+
+	[EVENT_TYPES.FRIENDS_LIST](data, socket) {
+		const {username} = socket.request.user
+		return this.UserModel.findOne({username}).exec()
+			.then(user => socket.emit(EVENT_TYPES.FRIENDS_LIST, user.friends))
+			.catch(err => logger.error(err))
```

Comment:
Will it throw error, or send empty array to user if nothing has been found?
I'd suggest to ensure that the user will receive response in such case.

Reply:
ok